docs/src/appendix/scarb-toml.md

```diff
@@ -0,0 +1,245 @@
+See [`unstable-add-statements-code-locations-debug-info`](https://docs.swmansion.com/scarb/docs/reference/manifest.html#unstable-add-statements-code-locations-debug-info) in Scarb docs.
+
+```toml
+[profile.dev.cairo]
+unstable-add-statements-code-locations-debug-info = true
+```
+
+#### `unstable-add-statements-functions-debug-info`
+See [`unstable-add-statements-functions-debug-info`](https://docs.swmansion.com/scarb/docs/reference/manifest.html#unstable-add-statements-functions-debug-info) in Scarb docs.
+```toml
+[profile.dev.cairo]
+unstable-add-statements-functions-debug-info = true
+```
+
+#### `inlining-strategy`
+See [`inlining-strategy`](https://docs.swmansion.com/scarb/docs/reference/manifest.html#inlining-strategy) in Scarb docs.
+```toml
+[profile.dev.cairo]
+inlining-strategy = "avoid"
+```
+
+#### Example of configuration which allows [coverage](https://foundry-rs.github.io/starknet-foundry/testing/coverage.html) report generation
+```toml
+[profile.dev.cairo]
+unstable-add-statements-code-locations-debug-info = true
+unstable-add-statements-functions-debug-info = true
+inlining-strategy = "avoid"
+```
+
+### `[features]`
+A package defines a set of named features in the `[features]` section of `Scarb.toml` file. Each defined feature can list other features that should be enabled with it. All fields are optional.
+```toml
+[features]
+# ...
+```
+
+#### `<feature-name>`
+The `<feature-name>` field specifies the name of the feature and list of other features that should be enabled with it.
+```toml
+[features]
+enable_for_tests = []
+```
+
+#### Example of `Scarb.toml` allowing conditional contracts compilation
+Firstly, define a contract in the src directory with a `#[cfg(feature: '<FEATURE_NAME>')]` attribute:
+```rust
+#[starknet::contract]
+#[cfg(feature: 'enable_for_tests')]
+mod MockContract {
+    // ...
+}
+```
+
+Then update Scarb.toml so it includes the following lines:
+```toml
+[features]
+enable_for_tests = []
+```
+
+See [features](https://docs.swmansion.com/scarb/docs/reference/conditional-compilation.html#features) in Scarb docs.
+
+### `[[target.starknet-contract]]`
+Scarb supports registering targets that are handled by Scarb extensions. Such targets are called external. All fields are optional.
+```toml
+[[target.starknet-contract]]
+# ...
+```
+
+#### `sierra`
+The `sierra` fields specifies whether Sierra codegen should be enabled.
+```toml
+[[target.starknet-contract]]
+sierra = true
+```
+
+#### `casm`
+The `casm` fields specifies whether Casm codegen should be enabled.
+```toml
+[[target.starknet-contract]]
+casm = true
+```
+
+See more about [targets](https://docs.swmansion.com/scarb/docs/reference/targets.html) in scarb docs.
+
+
+#### `build-external-contracts`
+The `build-external-contracts` accepts a list of strings, each of which is a reference to a contract defined in a dependency. The package that implements this contracts need to be declared as a dependency of the project in `[dependencies]`.
```

Comment:
This is a super formal description but doesn't really say what's the purpose of this in terms of forge.

Let's say that this is needed to use the contracts from dependencies in tests and also link to relevant scarb doc. You can keep the mention that these packages also need to be defined as dependencies, but this can be reworded.

Reply:
Hope this is better.

> The `build-external-contracts` allows to use contracts from your dependencies inside your tests. It accepts a list of strings, each of which is a reference to a contract defined in a dependency. You need to add dependency which  implements this contracts  to your `Scarb.toml`.

I've added a sentence which explains what this option actually does in regards of forge, should be clearer now.